src/ir/var.rs

```diff
@@ -130,6 +130,53 @@ fn default_macro_constant_type(value: i64) -> IntKind {
     }
 }
 
+/// Determines whether a set of tokens from a CXCursor_MacroDefinition
+/// represent a function-like macro. If so, calls the func_macro callback
+/// and returns `Err(ParseError::Continue)` to signal to skip further
+/// processing.  If conversion to UTF-8 fails (it is performed only where it
+/// should be infallible), then `Err(ParseError::Continue)` is returned as well.
+fn handle_function_macro(
+    tokens: &[clang::ClangToken],
+    callbacks: &dyn crate::callbacks::ParseCallbacks,
+) -> Result<(), ParseError> {
+    let mut spelled = tokens.into_iter().map(clang::ClangToken::spelling);
+    let boundary = spelled
+        .clone() // clone the iterator to avoid consuming elements
+        .position(|s| s == b")");
```

Comment:
nit: Instead of this, what do you think of doing `tokens.iter().position(|token| token.is_closing_paren())`, where `is_closing_paren()` checks the token kind to be `Punctuation` before calling `spelling()`, that allows to avoid doing useless work allocating strings for potentially large tokens for all the macros.

Reply:
That seems reasonable to me, so I did most of it, except for hoisting `is_closing_paren` up to `ClangToken`. I had already (in bbd1b1ac2d63ab0a0d12a283408ebbfe046ec787) pulled `is_abutting` out of `ClangToken` since it seems fairly special-purpose and I now need it only for libclang 3.8, so I thought keeping things local to this function would be better, but let me know if you think differently.